nannyml_cloud_sdk/monitoring/custom_metric.py

```diff
@@ -0,0 +1,218 @@
+
+
+_READ_CUSTOM_METRIC = gql("""
+    query readMetric($id: Int!) {
+        monitoring_metric(metricId: $id) {
+            ...MetricDetails
+        }
+    }
+""" + _CUSTOM_METRIC_DETAILS_FRAGMENT)
+
+
+_CREATE_CUSTOM_METRIC = gql("""
+    mutation createMetric($input: CreateMetricInput!) {
+        create_monitoring_metric(metric: $input) {
+            ...MetricDetails
+        }
+    }
+""" + _CUSTOM_METRIC_DETAILS_FRAGMENT)
+
+
+_DELETE_MONITORING_METRIC = gql("""
+    mutation deleteMetric($id: Int!) {
+        delete_monitoring_metric(metricId: $id) {
+            ...MetricDetails
+        }
+    }
+""" + _CUSTOM_METRIC_DETAILS_FRAGMENT)
+
+
+class CustomMetric:
+    """Class for interacting with custom metrics in NannyML Cloud."""
+
+    @classmethod
+    def list(cls, name: Optional[str] = None, problem_type: Optional[ProblemType] = None) -> List[CustomMetricSummary]:
+        """List defined custom metrics.
+
+        Args:
+            name: Optional name filter.
+            problem_type: Optional problem type filter.
+
+        Returns:
+            List of custom metrics that match the provided filter criteria.
+        """
+        res = execute(_LIST_CUSTOM_METRICS, {
+            'filter': {
+                'names': None if name is None else [name],
+                'problemTypes': None if problem_type is None else [problem_type],
+            }
+        })['monitoring_metrics']
+
+        return res
+
+    @classmethod
+    def get(cls, metric_id: int) -> CustomMetricDetails:
+        """Get details of a custom metric.
+
+        Args:
+            metric_id: Unique identifier of the custom metric.
+
+        Returns:
+            Details of the custom metric.
+        """
+        return execute(_READ_CUSTOM_METRIC, {'id': metric_id})['monitoring_metric']
+
+    @overload
+    @classmethod
+    def create(
+        cls,
+        name: str,
+        description: str,
+        problem_type: Literal["REGRESSION"],
+        *,
+        loss_function: str,
+        aggregation_function: str,
+        lower_value_limit: Optional[float] = None,
+        upper_value_limit: Optional[float] = None,
+    ) -> CustomMetricDetails:
+        ...
+
+    @overload
+    @classmethod
+    def create(
+        cls,
+        name: str,
+        description: str,
+        problem_type: Literal["BINARY_CLASSIFICATION", "MULTICLASS_CLASSIFICATION"],
+        *,
+        calculate_function: str,
+        estimate_function: Optional[str] = None,
+        lower_value_limit: Optional[float] = None,
+        upper_value_limit: Optional[float] = None,
+    ) -> CustomMetricDetails:
+        ...
+
+    @classmethod
+    def create(
+        cls,
+        name: str,
+        description: str,
+        problem_type: ProblemType,
+        calculate_function: Optional[str] = None,
+        estimate_function: Optional[str] = None,
+        loss_function: Optional[str] = None,
+        aggregation_function: Optional[str] = None,
```

Comment:
Oh almost forgot, do we want to also accept function objects here? The trick we had in mind using `inspect.getsource`

Reply:
Man, I'm not on it today. Excellent point